SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/SpatialComponentTest/SpatialComponentTest.cpp

```diff
@@ -0,0 +1,278 @@
+		AddStep(TEXT("Replicated Dynamic Actor Spawned On Same Server - Set owner to client 1"), FWorkerDefinition::Server(1), nullptr,
+				[this]() {
+					AController* PlayerController =
+						Cast<AController>(GetFlowController(ESpatialFunctionalTestWorkerType::Client, 1)->GetOwner());
+					DynamicReplicatedActor->SetOwner(PlayerController);
+					FinishStep();
+				});
+
+		AddStep(
+			TEXT("Replicated Dynamic Actor Spawned On Same Server - Verify components"), FWorkerDefinition::AllWorkers, nullptr, nullptr,
+			[this](float DeltaTime) {
+				Timer -= DeltaTime;
+				if (Timer <= 0)
+				{
+					// FinishStep();
+					CheckComponents(DynamicReplicatedActor, 1, 1, 0);
+				}
+			},
+			5.0f);
+
+		// Client 2 to get ownership of the dynamic actor - then check components
+		AddStep(TEXT("Replicated Dynamic Actor Spawned On Same Server - Set owner to client 2"), FWorkerDefinition::Server(1), nullptr,
+				[this]() {
+					AController* PlayerController =
+						Cast<AController>(GetFlowController(ESpatialFunctionalTestWorkerType::Client, 2)->GetOwner());
+					DynamicReplicatedActor->SetOwner(PlayerController);
+					FinishStep();
+				});
+
+		AddStep(
+			TEXT("Replicated Dynamic Actor Spawned On Same Server - Verify components"), FWorkerDefinition::AllWorkers, nullptr, nullptr,
+			[this](float DeltaTime) {
+				Timer -= DeltaTime;
+				if (Timer <= 0)
+				{
+					CheckComponents(DynamicReplicatedActor, 1, 2, 1);
+				}
+			},
+			5.0f);
+
+		AddStepFromDefinition(ReplicatedDestroyStepDefinition, FWorkerDefinition::Server(1));
+	}
+
+	// Replicated Dynamic Actor Spawned On Different Server. Server 1 should have Authority on BeginPlay, Server 2 on Tick
+	{
+		AddStep(TEXT("Replicated Dynamic Actor Spawned On Different Server - Spawn"), FWorkerDefinition::Server(1), nullptr, [this]() {
+			ASpatialComponentTestReplicatedActor* Actor =
+				GetWorld()->SpawnActor<ASpatialComponentTestReplicatedActor>(Server2Position, FRotator::ZeroRotator);
+			CrossServerSetDynamicReplicatedActor(Actor);
+			FinishStep();
+		});
+
+		AddStep(
+			TEXT("Replicated Dynamic Actor Spawned On Different Server - Verify components "
+				 "on Tick"),
+			FWorkerDefinition::AllWorkers, nullptr, nullptr,
+			[this](float DeltaTime) {
+				Timer -= DeltaTime;
+				if (Timer <= 0)
+				{
+					CheckComponentsCrossServer(DynamicReplicatedActor, 1, 2);
+				}
+			},
+			5.0f);
+
+		AddStepFromDefinition(ReplicatedDestroyStepDefinition, FWorkerDefinition::Server(2));
+	}
+}
+
+void ASpatialComponentTest::GetLifetimeReplicatedProps(TArray<FLifetimeProperty>& OutLifetimeProps) const
+{
+	Super::GetLifetimeReplicatedProps(OutLifetimeProps);
+
+	DOREPLIFETIME(ASpatialComponentTest, DynamicReplicatedActor);
+}
+
+// Checks the number of components on the servers when an actor does not migrate
+void ASpatialComponentTest::CheckComponents(ASpatialComponentTestActor* Actor, int ExpectedServerId, int ExpectedClient1ComponentCount,
+											int ExpectedClient2ComponentCount)
+{
```

Comment:
Minor: This function has a lot of branches for specific clients and servers, is it worth splitting these in steps that targets the servers and two for each client for readability? The client code can maybe done inline in the lambda's and this function can be left for servers?

Reply:
Maybe this is a personal preference of steps over branches. This function is called 4 times so we would need 4 extra steps if we move out the client code. Now I have used the RequireTrue this function is now more concise and easier to follow so perhaps it's better to leave the server and client checks together?

Reply:
I can see your point, still have a slight preference for my solution but its a style thing, fine as is : ).

Reply:
A follow on task that I am working on now is currently extending this test and I will probably split the server and client checks as it's getting overly complicated now. :-)